sdks/python/src/opik/evaluation/metrics/heuristics/bleu.py

```diff
@@ -0,0 +1,218 @@
+from typing import Any, List, Union, Optional
+
+try:
+    import nltk
+    from nltk.translate.bleu_score import sentence_bleu, corpus_bleu, SmoothingFunction
+except ImportError:
+    nltk = None
+
+from opik.evaluation.metrics import base_metric, score_result
+
+
+class BLEU(base_metric.BaseMetric):
+    """
+    BLEU metric relying on the `nltk.translate.bleu_score` implementation.
+    If `nltk` is not installed, this class will raise an ImportError upon instantiation.
+    """
+
+    def __init__(
+        self,
+        name: str = "bleu_metric",
+        track: bool = True,
+        n_grams: int = 4,
+        smoothing_method: str = "method1",
+        weights: Optional[List[float]] = None,
+    ):
+        """
+        :param name: Name for this metric instance.
+        :param track: Whether or not this metric is tracked (depends on your system).
+        :param n_grams: Up to which n-gram order to use (1 through n_grams).
+        :param smoothing_method: One of NLTK's SmoothingFunction methods (e.g., "method0", "method1", "method2", etc.).
```

Comment:
Let's add a link in the docstrings https://www.nltk.org/api/nltk.translate.bleu_score.html#nltk.translate.bleu_score.SmoothingFunction
so that people could understand those options easier.